api/http/Server.go

```diff
@@ -38,7 +46,7 @@ func (s *Server) Serve(wait *sync.WaitGroup) error {
 			err = http.ListenAndServeTLS(
 				":"+s.ServerConfig.HTTPSPort,
 				s.ServerConfig.TLSConfig.CertPath, //TODO: Needs sanity checking.
-				s.ServerConfig.TLSConfig.KeyPath, mux)
+				s.ServerConfig.TLSConfig.KeyPath, handler)
```

Comment:
I'd love to see some tests.  

Also: Was it intentional to add it only when SSL is set up and not [here](https://github.com/RedHatInsights/authz/pull/90/files#diff-db6116b4f5c82f4b81e77379964a2c6324130b00acc1ab9e1deacfa78d9b9a41R58) also? If so, I am not sure it is the best idea. A different approach could be to define the AllowedOrigins ( + perhaps Headers/Methods) in a configuration and load this for both.